pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/service/CookieUtils.scala

```diff
@@ -16,48 +25,39 @@ object CookieUtils {
       s"&expires=${authUser.expires}" +
       s"&multifactor=${authUser.multiFactor}"
 
-  private[service] def deserializeAuthenticatedUser(serializedForm: String): AuthenticatedUser = {
+  private[service] def deserializeAuthenticatedUser(serializedForm: String): Option[AuthenticatedUser] = {
     val data = serializedForm
       .split("&")
       .map(_.split("=", 2))
       .map{p => p(0) -> p(1)}
       .toMap
 
-    AuthenticatedUser(
-      user = User(data("firstName"), data("lastName"), data("email"), data.get("avatarUrl")),
-      authenticatingSystem = data("system"),
-      authenticatedIn = Set(data("authedIn").split(",").toSeq :_*),
-      expires = data("expires").toLong,
-      multiFactor = data("multifactor").toBoolean
+    for {
+      firstName <- data.get("firstName")
+      lastName <- data.get("lastName")
+      email <- data.get("email")
+      system <- data.get("system")
+      authedIn <- data.get("authedIn")
+      expires <- data.get("expires")
+      multifactor <- data.get("multifactor")
+    } yield AuthenticatedUser(
+      user = User(firstName, lastName, email, data.get("avatarUrl")),
+      authenticatingSystem = system,
+      authenticatedIn = Set(authedIn.split(",").toSeq :_*),
+      expires = expires.toLong,
```

Comment:
The return type of `deserializeAuthenticatedUser` has been changed to `Option[AuthenticatedUser]` in this branch. Is the intention that all foreseeable errors should therefore lead to a `None`, rather than throwing? If so, do we need to change how we're handling the `.toLong` call, given that it can throw?

Reply:
Good point! Addressed in f40aa8d7263952b5499070d3f01b2b4403821939.

Reply:
Whoops - ah, but it [fails](https://github.com/guardian/pan-domain-authentication/actions/runs/10321653497/job/28575045968#step:6:21) because [we have to compile under Scala 2.12](https://github.com/guardian/pan-domain-authentication/issues/149):

```
[error] /home/runner/work/pan-domain-authentication/pan-domain-authentication/pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/service/CookieUtils.scala:41:48: value toLongOption is not a member of String
[error]       expires <- data.get("expires").flatMap(_.toLongOption)
[error]                                                ^
[error] /home/runner/work/pan-domain-authentication/pan-domain-authentication/pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/service/CookieUtils.scala:42:56: value toBooleanOption is not a member of String
[error]       multiFactor <- data.get("multifactor").flatMap(_.toBooleanOption)
[error]                                                        ^
[error] two errors found
```

...I'll make something Scala 2.12 compliant...

Reply:
> ...I'll make something Scala 2.12 compliant...

Done with 998db1f097e51ee2d231ce23edcc7536384c4be3!